contracts/moda-contracts/src/interfaces/IModaRegistry.sol

```diff
@@ -2,47 +2,67 @@
 pragma solidity 0.8.21;
 
 interface IModaRegistry {
+    event CatalogMembershipChanged(string name, address indexed membership);
+    event CatalogRegistered(string name, address indexed catalog);
+    event CatalogUnregistered(string name, address indexed catalog);
+
+    event ManagerAdded(address indexed artist, address indexed manager);
+    event ManagerRemoved(address indexed artist, address indexed manager);
+
+    event TreasuryFeeChanged(uint256 oldFee, uint256 newFee);
+    event TreasuryChanged(address oldTreasury, address newTreasury);
+
     struct Catalog {
         string name;
-        address catalogAddress;
+        address catalog;
+        address membership;
     }
 
     /// Membership
 
     /**
      * @dev Checks the user is registered to a particular catalog
-     * @param catalogName The name of the catalog
-     * @param user The address of the user to check 
+     * @param index the index of the catalog
+     * @param user The address of the user to check
+     */
+    function isMember(uint256 index, address user) external returns (bool);
+
+    /**
+     * @dev sets a membership contract for a particular catalog
```

Comment:
Let's use consistent casing for the comments. These will generate documentation when we are done

Reply:
"Sets" vs "sets"